lib/src/realtime_subscription.dart

```diff
@@ -134,7 +134,7 @@ class RealtimeSubscription {
     void onClose() {
       socket.log('channel', 'leave $topic');
       trigger(ChannelEvents.close.eventName(),
-          payload: 'leave', ref: joinRef());
+          payload: {'type': 'leave'}, ref: joinRef());
```

Comment:
@phamhieu 
So the reason why the exception was being thrown was because here, the code is passing `leave` as String to trigger, and it goes [here](https://github.com/supabase/realtime-dart/blob/main/lib/src/realtime_subscription.dart#L201), which makes sense that the exception is thrown. Let me know if you prefer a different way of fixing this as my fix loses consistency with the js library.
https://github.com/supabase/realtime-js/blob/master/src/RealtimeSubscription.ts#L129

Also, would it make sense to change the type of `payload` to `Map` instead of dynamic? 
https://github.com/supabase/realtime-dart/blob/main/lib/src/realtime_subscription.dart#L181
Other than this `leave`, there is no where else that was passing `payload` as String in the code, but from the server could it be anything other than `Map`?

Reply:
@dshukertjr it seems to be a side effect of this commit https://github.com/supabase/realtime-dart/commit/dafdd9e35e07bf7f76155b26133ec9f1fd9ca48e#diff-87c3ec4317e9e9ddc1e314440d04bab1b0bba90a4b5c05e16e8b2e5abd11bdab

Reply:
> Or is leave not supposed to be a string on js library either?

good catch @dshukertjr. I think you are right.

Reply:
@dshukertjr can you also add a guard check here to prevent future crashing bug

https://github.com/supabase/realtime-dart/blob/731b8850be0754b7d741b7ee0019c6a997fb7392/lib/src/realtime_subscription.dart#L200-L202

Reply:
@phamhieu Sounds good, I will add it right now!

Reply:
Thank you 👍🚀